certs/pki/vault.go

```diff
@@ -98,44 +91,29 @@ func NewVaultClient(token, host, path, role string) (Agent, error) {
 		role:      role,
 		path:      path,
 		client:    client,
-		issueURL:  "/" + apiVer + "/" + path + "/" + issue + "/" + role,
-		readURL:   "/" + apiVer + "/" + path + "/" + cert + "/",
-		revokeURL: "/" + apiVer + "/" + path + "/" + revoke,
+		issueURL:  "/" + path + "/" + issue + "/" + role,
+		readURL:   "/" + path + "/" + cert + "/",
+		revokeURL: "/" + path + "/" + revoke,
 	}
 	return &p, nil
 }
 
-func (p *pkiAgent) IssueCert(cn string, ttl, keyType string, keyBits int) (Cert, error) {
+func (p *pkiAgent) IssueCert(cn, ttl string) (Cert, error) {
 	cReq := certReq{
 		CommonName: cn,
-		TTL:        ttl,
-		KeyBits:    keyBits,
-		KeyType:    keyType,
-	}
-
-	r := p.client.NewRequest("POST", p.issueURL)
-	if err := r.SetJSONBody(cReq); err != nil {
-		return Cert{}, err
-	}
-
-	resp, err := p.client.RawRequest(r)
-	if resp != nil {
-		defer resp.Body.Close()
+		TTL:        fmt.Sprintf("%sh", ttl),
```

Comment:
Do we need to add `h` prefix to ttl ?
Why the request can we with any prefix like `h`,`m`, `s`